core/src/ast/utils.rs

```diff
@@ -21,3 +24,224 @@ pub fn get_doc_lines(attrs: &[Attribute]) -> String {
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, StructField)] },
+            "https://docs.rs/std/latest/std/foo/struct.bar.html#structfield.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, Enum)] },
+            "https://docs.rs/std/latest/std/foo/bar/enum.batz.html",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, EnumVariant)] },
+            "https://docs.rs/std/latest/std/foo/enum.bar.html#variant.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, EnumVariantField)] },
+            "https://docs.rs/std/latest/std/enum.foo.html#variant.bar.field.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, Trait)] },
+            "https://docs.rs/std/latest/std/foo/bar/trait.batz.html",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, FnInStruct)] },
+            "https://docs.rs/std/latest/std/foo/struct.bar.html#method.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, FnInEnum)] },
+            "https://docs.rs/std/latest/std/foo/enum.bar.html#method.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, FnInTrait)] },
+            "https://docs.rs/std/latest/std/foo/trait.bar.html#tymethod.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, DefaultFnInTrait)] },
+            "https://docs.rs/std/latest/std/foo/trait.bar.html#method.batz",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, Fn)] },
+            "https://docs.rs/std/latest/std/foo/bar/fn.batz.html",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, Mod)] },
+            "https://docs.rs/std/latest/std/foo/bar/batz/index.html",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, Constant)] },
+            "https://docs.rs/std/latest/std/foo/bar/constant.batz.html",
+        ),
+        (
+            syn::parse_quote! { #[diplomat::rust_link(std::foo::bar::batz, Macro)] },
+            "https://docs.rs/std/latest/std/foo/bar/macro.batz.html",
+        ),
+    ];
+
+    let mut base_urls = HashMap::new();
+
+    for (attr, expected) in test_cases.clone() {
+        assert_eq!(get_rust_link(&[attr]).unwrap().http(&base_urls), expected,);
+    }
+
+    base_urls.insert("std".to_string(), "http://std-docs.biz/".to_string());
+
+    assert_eq!(
+        get_rust_link(&[test_cases[0].0.clone()])
+            .unwrap()
+            .http(&base_urls),
+        "http://std-docs.biz/std/foo/bar/struct.batz.html"
+    );
+}
```

Comment:
nit: i'd like some insta snapshot tests as well

Reply:
For this particular method? There are snapshot tests for `rust_link`